testing/camunda-process-test-java/src/main/java/io/camunda/process/test/api/mock/JobWorkerMock.java

```diff
@@ -30,34 +32,50 @@
 public interface JobWorkerMock {
 
   /** Configures the mock worker to complete jobs without any variables. */
-  void thenComplete();
+  FinalizedJobWorkerMock thenComplete();
 
   /**
    * Configures the mock worker to complete jobs with the specified variables.
    *
    * @param variables the variables to include when completing the job.
    */
-  void thenComplete(Map<String, Object> variables);
+  FinalizedJobWorkerMock thenComplete(Map<String, Object> variables);
 
   /**
    * Configures the mock worker to throw a BPMN error with the specified error code.
    *
    * @param errorCode the error code to throw.
    */
-  void thenThrowBpmnError(String errorCode);
+  FinalizedJobWorkerMock thenThrowBpmnError(String errorCode);
 
   /**
    * Configures the mock worker to throw a BPMN error with the specified error code and variables.
    *
    * @param errorCode the error code to throw.
    * @param variables the variables to include when throwing the error.
    */
-  void thenThrowBpmnError(String errorCode, Map<String, Object> variables);
+  FinalizedJobWorkerMock thenThrowBpmnError(String errorCode, Map<String, Object> variables);
 
   /**
    * Configures the mock worker with a custom job handler.
    *
    * @param jobHandler the custom job handler to use for processing jobs.
    */
-  void withHandler(JobHandler jobHandler);
+  FinalizedJobWorkerMock withHandler(JobHandler jobHandler);
+
+  interface FinalizedJobWorkerMock {
```

Comment:
:wrench: I suggest renaming the `JobWorkerMock` into `JobWorkerMockBuilder` and `FinalizedJobWorkerMock` into `JobWorkerMock`.

Since the mock is under construction until `then*()` or `withHandler()` is called, the name `builder` fits.

Renaming the interface is fine. I assume that nobody referenced the interface directly.

Reply:
:wrench: Add JavaDoc on the interface.